### ISSUE TITLE
Incorrect implementation of Batch Renorm?

### ISSUE THREAD

**Issue by waleedka** [ppwwyyxx/tensorpack#360]
I've been checking your implementation of Batch Re-Normalization, and I think there is an error that you might want to look into. 

To set a baseline, this is how the original paper defines batch renormalization:

```Python
normed_x = (x - batch_mean) / batch_std    # standard BN
normed_x = normed_x * r + d                # The batch renorm correction
normed_x = normed_x * gamma + beta         # final scale and bias
```

In other words, apply BN first, then apply the r/d correction, then apply gamma/beta last. But, in batch_norm.py, it applies BN, then the gamma/beta (both operations are done using tf.nn.fused_batch_norm in this line):

https://github.com/ppwwyyxx/tensorpack/blob/master/tensorpack/models/batch_norm.py#L209

Then it applies the r/d correction last in:
https://github.com/ppwwyyxx/tensorpack/blob/master/tensorpack/models/batch_norm.py#L220

It's possible that I'm missing something here or misunderstood the paper. If so, please do correct me.

**Comment by ppwwyyxx** [comment]
You're right.

**Comment by ppwwyyxx** [comment]
Btw I'm going to fix it by just letting it call `tf.layers.batch_norm` with renorm option, since wrapping symbolic functions to layers is not a focus of this project. You can also just call `tf.layers` instead of tensorpack layers.

**Comment by ppwwyyxx** [comment]
Thanks a lot!
